internal/webserver/utils-ip.go

```diff
@@ -0,0 +1,219 @@
+		// 198.18.0.0/15
+		{
+			IP:   []byte{198, 18, 0, 0},
+			Mask: []byte{255, 254, 0, 0},
+		},
+		// RFC 3171
+		// 224.0.0.0/4
+		{
+			IP:   []byte{224, 0, 0, 0},
+			Mask: []byte{240, 0, 0, 0},
+		},
+		// RFC 1112
+		// 240.0.0.0/4
+		{
+			IP:   []byte{240, 0, 0, 0},
+			Mask: []byte{240, 0, 0, 0},
+		},
+		// RFC 919 Section 7
+		// 255.255.255.255/32
+		{
+			IP:   []byte{255, 255, 255, 255},
+			Mask: []byte{255, 255, 255, 255},
+		},
+		// RFC 6598
+		// 100.64.0.0/10
+		{
+			IP:   []byte{100, 64, 0, 0},
+			Mask: []byte{255, 192, 0, 0},
+		},
+	}
+	// Sourced from https://www.iana.org/assignments/iana-ipv6-special-registry/iana-ipv6-special-registry.xhtml
+	// where Global, Source, or Destination is False
+	privateV6Networks = []net.IPNet{
+		parseCidr("::/128", "RFC 4291: Unspecified Address"),
+		parseCidr("::1/128", "RFC 4291: Loopback Address"),
+		parseCidr("::ffff:0:0/96", "RFC 4291: IPv4-mapped Address"),
+		parseCidr("100::/64", "RFC 6666: Discard Address Block"),
+		parseCidr("2001::/23", "RFC 2928: IETF Protocol Assignments"),
+		parseCidr("2001:2::/48", "RFC 5180: Benchmarking"),
+		parseCidr("2001:db8::/32", "RFC 3849: Documentation"),
+		parseCidr("2001::/32", "RFC 4380: TEREDO"),
+		parseCidr("fc00::/7", "RFC 4193: Unique-Local"),
+		parseCidr("fe80::/10", "RFC 4291: Section 2.5.6 Link-Scoped Unicast"),
+		parseCidr("ff00::/8", "RFC 4291: Section 2.7"),
+		// We disable validations to IPs under the 6to4 anycase prefix because
+		// there's too much risk of a malicious actor advertising the prefix and
+		// answering validations for a 6to4 host they do not control.
+		// https://community.letsencrypt.org/t/problems-validating-ipv6-against-host-running-6to4/18312/9
+		parseCidr("2002::/16", "RFC 7526: 6to4 anycast prefix deprecated"),
+	}
+)
+
+// parseCidr parses the predefined CIDR to `net.IPNet` that consisting of IP and IPMask.
+func parseCidr(network string, comment string) net.IPNet {
+	_, subNet, err := net.ParseCIDR(network)
+	if err != nil {
+		panic(fmt.Sprintf("error parsing %s (%s): %s", network, comment, err))
+	}
+	return *subNet
+}
+
+// isPrivateV4 checks whether an `ip` is private based on whether the IP is in the private CIDR range.
+func isPrivateV4(ip net.IP) bool {
+	for _, subNet := range privateNetworks {
+		if subNet.Contains(ip) {
+			return true
+		}
+	}
+	return false
+}
+
+// isPrivateV6 checks whether an `ip` is private based on whether the IP is in the private CIDR range.
+func isPrivateV6(ip net.IP) bool {
+	for _, subNet := range privateV6Networks {
+		if subNet.Contains(ip) {
+			return true
+		}
+	}
+	return false
+}
+
+// IsPrivateIP check IPv4 or IPv6 address according to the length of byte array
+func IsPrivateIP(ip net.IP) bool {
+	if ip4 := ip.To4(); ip4 != nil {
+		return isPrivateV4(ip4)
+	}
+	return len(ip) == IPv6Len && isPrivateV6(ip)
+}
+
+// IsIpValidAndPublic is a helper function check if an IP address is valid and public.
+func IsIpValidAndPublic(ipAddr string) bool {
```

Comment:
nit: `IsIPValidAndPublic`

Reply:
Ok, I will fix all code-style issues in minutes.